actions/generate_changelog.go

```diff
@@ -0,0 +1,144 @@
+package actions
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"text/template"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+const (
+	changelogTpl string = `{{.OldRelease}} -> {{.NewRelease}}
+
+# Features
+
+{{range .Features}} - {{.}}
+{{else}}No new features for this release.
+{{end}}
+
+# Fixes
+
+{{range .Fixes}} - {{.}}
+{{else}}No bug fixes for this release.
+{{end}}
+
+# Documentation
+
+{{range .Documentation}} - {{.}}
+{{else}}No new documentation for this release.
+{{end}}
+
+# Maintenance
+
+{{range .Maintenance}} - {{.}}
+{{else}}No maintenance required for this release.
+{{end}}`
+)
+
+type Changelog struct {
+	OldRelease string
+	NewRelease string
+	Features []string
+	Fixes []string
+	Documentation []string
+	Maintenance []string
+}
+
+// GenerateChangelog is the CLI action for creating an aggregated changelog from all of the Deis Workflow repos.
+func GenerateChangelog(client *github.Client, dest io.Writer) func(*cli.Context) {
+	return func(c *cli.Context) {
+		changelog := &Changelog{
+			OldRelease: c.Args().Get(0),
+			NewRelease: c.Args().Get(1),
+		}
+		if changelog.OldRelease == "" || changelog.NewRelease == "" {
+			log.Fatal("Usage: generate-changelog <old-release> <new-release>")
+		}
+		if err := generateChangelog(client, changelog); err != nil {
+			log.Fatalf("could not generate changelog: %s", err)
+		}
+		t := template.Must(template.New("changelog").Parse(changelogTpl))
+		err := t.Execute(dest, changelog)
+		if err != nil {
+			log.Fatalf("could not template changelog: %s", err)
+		}
+	}
+}
+
+func generateChangelog(client *github.Client, changelog *Changelog) error {
+	var wg sync.WaitGroup
+	done := make(chan bool)
+	errCh := make(chan error)
+	defer close(errCh)
+	for _, name := range repoNames {
+		wg.Add(1)
+		go func(name string) {
+			defer wg.Done()
+			commitCompare, resp, err := client.Repositories.CompareCommits("deis", name, changelog.OldRelease, changelog.NewRelease)
+			if err != nil {
+				if resp.StatusCode == http.StatusNotFound {
+					log.Printf("tag does not exist for this repo; skipping %s", name)
+					return
+				}
+				errCh <- fmt.Errorf("could not compare commits %s and %s: %s", changelog.OldRelease, changelog.NewRelease, err)
```

Comment:
if 2 goroutines error, the second one will never complete the send. one possible fix is to consume and return a slice of all errors returned. wouldn't say this is blocking for this PR because the process will exit immediately on the first error though